### DISCUSSION TITLE
Looking for developers and artists!!!

### DISCUSSION THREAD

**Discussion by C0rupted** [C0rupted/Vitrix#26] [General]
If you've clicked this post, you most likely want to apply to become a developer of Vitrix. If you would like to do so, copy the following template, fill it out and comment it below:
```
Field of expertise:    (what specific area are you best at doing? modelling or code, textures or testing?)
Online time:           (when you are available to help contribute to Vitrix)
Reason:                (why are you applying to help Vitrix)
```
Filling out the following will get you entered onto the list, and confirmed developers will be announced soon!

**Comment by ghost** [comment]
I would love to be involved, but with other responsibilities, it will be hard for me to be present 24x7. Can I please work at my own pace until I can sort out 'income-related' difficulties?
```
Expertise: Core python, Django, story-writing.
Time: UTC 3.30 AM TO 11.30 AM
Reason: I was looking for some open-source python project so I can utilize my python skills. I can draw and model but that isn't my strong suit therefore working alone would be hard for me.
```
I think I can accumulate more knowledge and chance to meet cool people through this project and I hope to be a useful and supportive team member. I intend for no personal gain except to hone my skills. I only request some moral support and open mindfulness from other members as I have never worked on a project with a team before.
Thank you and I wish Vitrix to be cherished.

**Reply by C0rupted** [reply]
Sure, these are the perfect kind of developers I am looking for. However, as Vitrix is largely made using Ursina Engine, you will also need to learn to use it, luckily, there are many useful tutorials available to do so! On the Reddit page, you stated that you aren't very good at github. [Here](https://www.youtube.com/watch?v=mVnZVw4KJnc) is a short tutorial, if you'd like to learn more, there is also [this](https://www.youtube.com/watch?v=RGOj5yH7evk) crash course.

**Reply by C0rupted** [reply]
You are now in the list, and confirmed devs will be announced soon!

**Comment by ghost** [comment]
I am watching them right now

**Reply by C0rupted** [reply]
If you have any idewas you'd like to propose, please mention them [here](https://github.com/ShadityZ/Vitrix/discussions/24)

**Comment by ghost** [comment]
It's my responsibility now. We must communicate daily and share our
thoughts and goals.
I have configured git on PC and forked the repo and am ready to go.
Although I have a question. *Every time you add new files to the project do
I have to fork everytime to update my repo?*

**Reply by C0rupted** [reply]
You might not know this as you are new to git, but once the roles have been announced, I will add the people to a list of collaborators that can commit to the repository directly, using a cloned version of the repo on your hard drive, and forking is unnecessary. TBH, I only just found out about that option my self. :)

**Comment by ghost** [comment]
Concern:
I checked build.py and saw a missingimport warning from

if operating_sys == "linux":
    from* venvctl *import* VenvCtl *
Now I want to ask that does it work without importing venvctl when executed
from linux or you have to install it manually? I am not knowledgeable in
linux sorry.

**Reply by C0rupted** [reply]
Yes. I seem to have the same problem myself until I went into the site packages and found that I had two versions of python installed, 3.9 and 3.10. However, VenvCTL automatically installs into 3.10, and by default, Visual Studio Code uses 3.9, so it says that it can't find the module until you switch your python interpreter. No idea why, but switch your interpreter seems to fix the issue. Keep in mind, you might have different versions installed compared to me, however both platforms seem to have this issue.

**Comment by ghost** [comment]
You posted in Issues that skybox isn't working.
But it's working in Windows. Check it out,
It's working perfectly without any issue.
[image: SharedScreenshot.png]

**Reply by C0rupted** [reply]
Yes. I have realised now. It seems to be a problem with the way that VS code handles project folders. I am working on a fix now. Updates coming in the issue

**Comment by davecarrijo** [comment]
I’m here to apply to the design role. I will answer the question in my lunch break.

**Reply by ghost** [reply]
Thank you to have you here brother.

**Reply by davecarrijo** [reply]
> Thank you to have you here brother.

ty for having me o/

**Comment by C0rupted** [comment]
I have been thinking, and I only have access to Windoes, Linux and a Raspberry Pi, and not a Mac device. If anybody has a Mac, and would be able to help test them, please reply to this message, and include which Mac you have (M1 or Intel).

**Comment by davecarrijo** [comment]
Expertise: Design[PS,IA,AF,ZBRUSH,CINEMA4D,BLENDER,painttoolSAI] Core python, Markdown, Sass, JS, godot
Time:UTC+1 3.30 AM TO 5.00 AM them 14pm to 17pm.
Reason: I was looking for some open-source python project so I can learn about python and how to work in a team.

**Reply by C0rupted** [reply]
Cool, you're now in the list! Team to be announced in 3 days!

**Comment by ghost** [comment]
That's excellent. But we need a better mode of communication sir. I need to
discuss so much with you .

**Reply by C0rupted** [reply]
Yes. I have looked at your repo, and I have found you and @davecarrijo have actually been pushing changes to the master branch. However, you should never push to master unless there is a major problem to fix. I understand that your new to github, so you might not know this. Instead try pushing your commits to the dev branch instead, where they can be tested then merged at the end of the month like the April PR #9. 

I do agree with you on a new method of communication. Maybe we could use Discord?

**Reply by davecarrijo** [reply]
Actually I has and yes, you're correct, I'm new with git but already reading about it.

**Comment by ghost** [comment]
>
> Yes. I have looked at your repo, and I have found you a d @davecarrijo
> <https://github.com/davecarrijo> have actually been pushing changes to
> the master branch.

I am really sorry. I am very much unaware of GitHub procedures. I will not
make these mistakes anymore.

> Maybe we could use Discord?

I agree with you. Discord would be best because it has all the criteria we
need right now - voice channels, concept sharing, and easily manageable. We
need voice communication more than anything right now because there is no
goal set for this project.

**Comment by ghost** [comment]
*Inquiry:*
I have some confusion if you kindly can help it out.

   - Let's say five people are contributing to this project and everyone is
   committing simultaneously. Now how would you decide which version you need
   to test and which block of code needs to be untouched?
   - I have made changes to the folder structure of the *assets* folder.
   How to know that you're okay with it and all other contributors are
   continuing with it?
   - I have made a new bullet type which you have probably noticed requires
   a .obj file to work. For future reference, I would like to know if we are
   continuing with the .obj format or something else.
   - Currently, the main file runs from *menu.py.* Shouldn't we name it
   *main.py* so that we don't get into conflict later as we need to have a
   separate script for menu-related functions?

**Reply by C0rupted** [reply]
Okay, here are some answers to your questions:
- First of all, when adding something to Vitrix you should always make sure that you have the latest version, this can be achieved by running ```git pull```.
- I know that you have done so, yes. However, before alerting the change, please make a message on the soon to be discord to alert everybody about the change, however they should know if they updated to the latest version as I mentioned as I answered question 1.\
- Yes. For images and textures, we are sticking to .png, as the quality is decent, but the file isn't so large that it lags out your computer. For models, yes .obj is the main one, as it is more common and Ursina Engine seems to support it. (I know Ursina also supports .blend, but it requires blender to be installed and that may not be the case on all computers). Sound is referenced as .wav as not many libs in python support .mp3 and Ursina's built in Audio class seems to use .wav as well. 
- Yes. In fact, it is a part of the roadmap for the monthly PR, May 2022.

I hope that answers your questions!

**Reply by davecarrijo** [reply]
It did!

**Comment by C0rupted** [comment]
I'd like to welcome everybody to the Official Vitrix Discord Server:

https://discord.gg/Vpmwn7HEPp

NOTE FOR THE TEAM:     Please message me you discord usernames, so I can identify you and give you permission for the developer chat.

**Reply by davecarrijo** [reply]
Idk why, but the link is broken.I click and nothing.

**Reply by Evhann** [reply]
> Idk why, but the link is broken.I click and nothing.

https://discord.gg/z3PKve9Qk5

**Reply by C0rupted** [reply]
> 

Seems fine too me

**Comment by C0rupted** [comment]
## THE TEAM:
- **Head Designer and Artist** - @davecarrijo 
- **Developers** - @SandipanGuha and @ShadityZ
- **Tester** - @Chateauvisionn 

If anybody else applies, this list may change.

**Comment by C0rupted** [comment]
I have invited the team to collaborate on the main github repo. 

#### RULES
- Before making a commit, you must confirm it with the rest of the dev team.
- You must send all your commits to the ```dev``` branch unless the commit comes with a security fix, or fixes another major bug, and if possible also confirm with the dev team first.
- Follow the roadmap (will be discussed on the discord later), and don't start developing ideas without discussing it with the dev team first.

If you do not follow these rules or misuse your privilege, I must immediately remove you from the collaborators list.
I need consent from both @SandipanGuha and @davecarrijo .

**Reply by davecarrijo** [reply]
I consent to the rules.

**Reply by C0rupted** [reply]
Okay. Please join the discord server so I can give you developer role and you can join us in the developer chat channel. Also accept the invitation to be a collaborator, as github says I am still awaiting your response, you should be able to find it in your email

**Reply by davecarrijo** [reply]
Here is my discord ID waiolos#7564

**Comment by ghost** [comment]
I have read all the above-mentioned rules and I swear I will abide by them.
From now on I will push all commits to* dev* branch and implement new ideas
after thoroughly discussed in discord or GitHub discussion.
-Sandipan Guha ***@***.***>
-Kolkata, India
-Contact: (+91)8617219957

**Reply by C0rupted** [reply]
I have givenyou the developer role on the discord server and you should now be able to see and message in the developer chat channel. All further discussion can continue there

**Comment by Evhann** [comment]
Field of expertise:   Coding, testing
Online time:           13h-19h [UTC+2](https://time.is/en/UTC+2)
Reason:                I liked creating mods for the game and i want to contribute to the game itself and improve user experience

**Reply by C0rupted** [reply]
Okay! Your in the team! Adding you to the discord as well.

**Reply by Evhann** [reply]
i can't be here between 7:30 and 16:40 the monday, tuesday, thursday, friday 7:30 and 11:20: the wednesday (school)

**Comment by ghost** [comment]
*Communication-related:*
I would like to thank you for showing interest in me and I would love to
contribute as much as I can. But for that to happen we need a better
method of communication. I would like to propose creating a discord server
dedicated to our project but if that isn't much priority for you right now
then invite me on your personal server or join mine instead
*https://discord.com/invite/gsuhxc269d
<https://discord.com/invite/gsuhxc269d>*
so that we talk freely and share our words. It would also help us share
artworks, links, and documents. I am eagerly waiting for your response.
Thank you.

**Reply by davecarrijo** [reply]
Vitrix already have a discord server : https://discord.gg/mtPmH6My
